pkg/codegen/nodejs/gen_program_test.go

```diff
@@ -29,41 +30,51 @@ func nodejsCheck(t *testing.T, path string, dependencies codegen.StringSet) {
 	ex, err := executable.FindExecutable("yarn")
 	require.NoError(t, err, "Could not find yarn executable")
 	dir := filepath.Dir(path)
-	pkgs := nodejsPackages(dependencies)
+	pkgs := nodejsPackages(t, dependencies)
 	// We delete and regenerate package files for each run.
-	packageJSON := filepath.Join(dir, "package.json")
-	if err := os.Remove(packageJSON); !os.IsNotExist(err) {
-		require.NoError(t, err, "Failed to delete %s", packageJSON)
+	packageJSONPath := filepath.Join(dir, "package.json")
+	if err := os.Remove(packageJSONPath); !os.IsNotExist(err) {
+		require.NoError(t, err, "Failed to delete %s", packageJSONPath)
 	}
 	yarnLock := filepath.Join(dir, "yarn.lock")
 	if err := os.Remove(yarnLock); !os.IsNotExist(err) {
 		require.NoError(t, err, "Failed to delete %s", yarnLock)
 	}
 
-	err = integration.RunCommand(t, "link @pulumi/pulumi",
-		[]string{ex, "link", "@pulumi/pulumi"},
-		dir, &integration.ProgramTestOptions{})
-	require.NoError(t, err, "Failed to link @pulumi/pulumi")
-	for _, pkg := range pkgs {
-		err = integration.RunCommand(t, "yarn add and install",
-			[]string{ex, "add", pkg}, dir, &integration.ProgramTestOptions{})
-		require.NoError(t, err, "Could not install package: %q", pkg)
+	pkgInfo := npmPackage{
+		Dependencies: map[string]string{
+			"@pulumi/pulumi": "latest",
+		},
+		DevDependencies: map[string]string{
+			"@types/node": "latest",
+			"typescript":  "latest",
```

Comment:
Is it possible that those `latest` versions may produce flaky results over time?

Reply:
I can pin to the current version (latest at this moment). I don't think there will be meaningful flake from TS and node, but pinning will make sure.